application_sandbox/stencil/main.cpp

```diff
@@ -0,0 +1,486 @@
+    mirror_pipeline_->SetFrontFace(VK_FRONT_FACE_CLOCKWISE);
+    mirror_pipeline_->SetInputStreams(&cube_);
+    mirror_pipeline_->SetViewport(viewport());
+    mirror_pipeline_->SetScissor(scissor());
+    mirror_pipeline_->SetSamples(num_samples());
+    // Enable color blend
+    mirror_pipeline_->AddDynamicState(VK_DYNAMIC_STATE_BLEND_CONSTANTS);
+    mirror_pipeline_->AddAttachment(VkPipelineColorBlendAttachmentState{
+        VK_TRUE,                         // blendEnable
+        VK_BLEND_FACTOR_CONSTANT_COLOR,  // srcColorBlendFactor
+        VK_BLEND_FACTOR_ONE,             // dstColorBlendFactor
+        VK_BLEND_OP_ADD,                 // colorBlendOp
+        VK_BLEND_FACTOR_ONE,             // srcAlphaBlendFactor
+        VK_BLEND_FACTOR_ONE,             // dstAlphaBlendFactor
+        VK_BLEND_OP_MAX,                 // alphaBlendOp
+        VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT |
+            VK_COLOR_COMPONENT_B_BIT | VK_COLOR_COMPONENT_A_BIT
+        // colorWriteMask
+    });
+    // Need to test the stencil buffer. The reference and compare mask are to
+    // be set later dynamically. The real value used to compare to the stencil
+    // buffer is 'reference & compare mask'.
+    mirror_pipeline_->AddDynamicState(VK_DYNAMIC_STATE_STENCIL_REFERENCE);
+    mirror_pipeline_->AddDynamicState(VK_DYNAMIC_STATE_STENCIL_COMPARE_MASK);
+    mirror_pipeline_->DepthStencilState().stencilTestEnable = VK_TRUE;
+    mirror_pipeline_->DepthStencilState().front.compareOp = VK_COMPARE_OP_EQUAL;
+    // Disable depth test, so the reflection can be shown on the floor.
+    mirror_pipeline_->DepthStencilState().depthTestEnable = VK_FALSE;
+
+    mirror_pipeline_->Commit();
+
+    // Transformation data for viewing and cube/floor rotation.
+    camera_data_ = containers::make_unique<vulkan::BufferFrameData<CameraData>>(
+        data_->root_allocator, app(), num_swapchain_images,
+        VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT);
+
+    model_data_ = containers::make_unique<vulkan::BufferFrameData<ModelData>>(
+        data_->root_allocator, app(), num_swapchain_images,
+        VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT);
+
+    float aspect =
+        (float)app()->swapchain().width() / (float)app()->swapchain().height();
+    camera_data_->data().projection_matrix =
+        Mat44::FromScaleVector(mathfu::Vector<float, 3>{1.0f, -1.0f, 1.0f}) *
+        Mat44::Perspective(1.5708f, aspect, 0.1f, 100.0f);
+
+    model_data_->data().transform =
+        Mat44::FromTranslationVector(
+            mathfu::Vector<float, 3>{0.0f, 0.0f, -3.0f}) *
+        Mat44::FromRotationMatrix(Mat44::RotationX(3.14f * 0.2f));
+  }
+
+  virtual void InitializeFrameData(
+      CubeFrameData* frame_data, vulkan::VkCommandBuffer* initialization_buffer,
+      size_t frame_index) override {
+    // Initalize the depth stencil image and the image view.
+    VkImageCreateInfo depth_stencil_image_create_info = {
+        /* sType = */
```

Comment:
In general I try and avoid /* */-style comments. Since you can't nest them they make it hard to comment out chunks of the code for testing.